ovs/dal/dataobject.py

```diff
@@ -736,7 +739,7 @@ def delete(self, abandon=None, _hook=None):
                 self._persistent.apply_transaction(transaction)
                 successful = True
             except KeyNotFoundException as ex:
-                if ex.message != self._key:
+                if self._key not in ex.message:
```

Comment:
Aha, interesting, that was not on purpose. It might be a better idea though to change the pyrakoonstore to pass the field as message instead of the exception. ```field``` in there is the exception and not the field itself, so it's name is wrong.